### ISSUE TITLE
Recompile current LESS without saving it

### ISSUE THREAD

**Issue by RaptorCZ** [madskristensen/WebEssentials2013#1632]
Hi, is there any way how to force recompile current LESS (opened in editor or selected in solution view) without saving LESS file? Sometimes I need to just recompile whole LESS to apply changes from other imported LESS files (no auto resolve build) but without TFS checkout of current LESS.

All I'm looking for is option "Recompile this". There are "Recompile all" in WE menu, but it is not what I'm looking for.

**Comment by bradlis7** [comment]
This would probably be more of a headache, but in my case, detecting all parent less files and compiling them would be great (so if I edit buttons.less in Bootstrap, it compiles bootstrap.less). If that's too much work, then the context menu would be excellent.

**Comment by CodeChief** [comment]
Very annoying having to edit master files which should not be changed. I'm looking for a clean solution where the bootstrap source is auto-updated and only the variables are replaced. File time stamps are important and editing old files just to make them compile is a dirty workaround and makes them appear newer than the updated version! Surely it's a trivial task just to re-enable the menu item, or better use the Visual Studio build/custom tool action as other extensions do neatly. Please add either of those features (the second is the best as it supports auto-update).

**Comment by SLaks** [comment]
@CodeChief WE should automatically recompile all files that import any file you save (Auto-compile dependant files on save in options)

**Comment by CodeChief** [comment]
Yes I noticed after the first "edit" it does it again. Just a design/usability issue here. It's not at all intuitive to see the context menu with all disabled options and have to edit (save) a master file you don't want to change just to indirectly trigger the tool to do something it should have the option for directly.

By the way the correct method other extensions use is to select the build action or custom tool property to persist a developer's decision as to what should normally be done with one file or another in a project. You answered a similar post here claiming it will not allow the parent item to be build type "none" or "content" but that is wrong because that is exactly what the normal use case is, look at resource files, settings files and many other Visual Studio files types for example. The parent item and auto-generated (project linked items) can have their own build types.

What we have here is more like a custom tool. We are generating code off master files. The custom tool pattern also gives the user the build-in "run custom tool" option for manual override. Saving the file to cause a compilation is just wrong.

**Comment by advancedrei** [comment]
The whole point is, I want control over when an individual LESS file is recompiled. Meaning I should be able to trigger a recompilation of an individual file whenever I want. There are certain situations where I am debugging an ASP.NET app, and I change a LESS or TS file, and because the debugger is running, it won't recompile the file, even if I changed it and saved it. That is very irritating. I just want to be able to right-click a LESS file and select "compile", regardless of whether or not it thinks I should/need to.
